source/common/router/router.cc

```diff
@@ -452,7 +477,14 @@ void Filter::onUpstreamReset(UpstreamResetType type,
       body = "upstream connect error or disconnect/reset before headers";
     }
 
-    chargeUpstreamCode(code, upstream_host);
+    chargeUpstreamCode(code, upstream_host,
+                       reset_reason.valid() &&
+                           reset_reason.value() == Http::StreamResetReason::Overflow);
+    // If we had non-5xx but still have been reset by backend or timeout before
+    // starting response, we treat this as an error.
```

Comment:
I think we are talking past each other here. I was tying to say that the only time you actually need the following if check is in the alt-timeout-response-code case, otherwise it will always be 5xx, right? I was just suggesting making the comment explicitly say this because to the casual reader this is a little confusing as to why it's not always 5xx in this path.

Reply:
Got it.